### PR TITLE
clients/erigon: Dockerfile support to build Erigon from git or locally.

### PR DESCRIPTION
**Pull request by spencer-tb**
Requires: https://github.com/ethereum/hive/pull/767.

Adds clients/erigon/Dockerfile.git to build Erigon from any git branch.
Adds clients/erigon/Dockerfile.local to build Erigon from a local directory.

### CODE REVIEW

File: clients/erigon/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
 # The upstream erigon container uses a non-root user, but we need
 # to install additional commands, so switch back to root.
 USER root
 
-# Install script tools.
-RUN apk add --update bash curl jq
+RUN apk add --no-cache \
+    bash \
+    curl \
+    jq
```
Comment (holiman) [inline clients/erigon/Dockerfile:14]:
```suggestion
RUN apk add --no-cache bash curl jq
```

> The --no-cache option allows to not cache the index locally, which is useful for keeping containers small.
> Literally it equals apk update in the beginning and rm -rf /var/cache/apk/* in the end.

Nice, today I learned. Prefer one line though

File: clients/erigon/Dockerfile.git [hunk 2]
```diff
@@ -1,45 +1,54 @@
...
+COPY --from=builder /go/erigon/build/bin/* /usr/local/bin/
 
-# Create version.txt
-RUN /usr/local/bin/erigon --version | sed -e 's/erigon version \(.*\)/\1/' > /version.txt
+RUN apk add --no-cache \
+    bash \
+    curl \
+    jq \
+    libstdc++ \
+    libgcc
```
Comment (holiman) [inline clients/erigon/Dockerfile.git:37]: Why are thes (libstdc++, libgcc) needed in the "not-builder"?

**Review by holiman — COMMENTED** [review]
Looks mostly good to me

File: clients/erigon/Dockerfile.git (hunk 2 above)
Comment (spencer-tb) [reply clients/erigon/Dockerfile.git:37]: Ahh so I believe these are already included in the default Dockerfile base image with a pre-built Erigon. 

As we are building from source here (‘.git’) and also in ‘.local’ : in the “not-builder” stage where we are running the client, these dependencies are needed to run the client successfully

File: clients/erigon/Dockerfile (hunk 1 above)
Comment (spencer-tb) [reply clients/erigon/Dockerfile]: Made some small changes! 

This led down the path of trying to reduce the size of the builder stage only to realize it was redundant for a multi stage docker build so I also learned something :D

**Pull request merged**